graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala

```diff
@@ -155,6 +169,8 @@ object Pregel extends Logging {
       i += 1
     }
     messages.unpersist(blocking = false)
```

Comment:
Looks like unpersist is a protected method and we cannot access it from Pregel. Add a new public method to unpersist dataset to work around this.